ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs

```diff
@@ -177,6 +179,24 @@ data TraceForgeEvent blk tx
   -- See also <https://github.com/input-output-hk/ouroboros-network/issues/1462>
   | TraceBlockFromFuture SlotNo SlotNo
 
+  -- | Leadership check failed: the tip of the ImmDB inhabits a slot /after/
```

Comment:
Not "after the current slot" but "the current slot". The ImmutableDB truncates blocks from the future, but not from the present (= the current slot), see #1563.